### PR TITLE
Updating saved page status using KVO in the article VC

### PR DESCRIPTION
**Pull request by coreyfloyd**
What -  pull request!!

### CODE REVIEW

**Comment by coreyfloyd** [comment]
@bgerstle @montehurd merged in master to get travis stuff

**Comment by bgerstle** [comment]
-0.01%?!?!?! 😡😦😱😫😨😩😤
j/k will look later

File: Wikipedia/UI-V5/WMFArticleViewController.m [hunk 1]
```diff
@@ -129,6 +133,18 @@ - (WMFArticleFetcher*)articleFetcher {
     return _articleFetcher;
 }
 
+#pragma mark - Saved Pages KVO
+
+- (void)observeSavedPages {
+    [self.KVOControllerNonRetaining observe:self.savedPages keyPath:WMF_SAFE_KEYPATH(self.savedPages, entries) options:0 block:^(id observer, id object, NSDictionary* change) {
+        [self updateSavedButtonState];
```
Comment (bgerstle) [inline Wikipedia/UI-V5/WMFArticleViewController.m]: womp womp, retain cycle. please use `observer` arg for safety

**Comment by coreyfloyd** [comment]
dios mio!

**Comment by bgerstle** [comment]
:-1: cell save status doesn't update properly:

#### repro steps
1. tap on "recents" tab
2. search for "planet"
3. tap "Planet" page
4. Swipe to dismiss & tap "Cancel" to dismiss search
5. Note that it's in the list, un-saved
6. Search for "Planet" again
7. Tap "save" button on "Planet" page
8. Go back to list

#### Expected results
Heart icon of page in recents is filled

#### Actual results
Heart icon is empty

**Comment by coreyfloyd** [comment]
@bgerstle building now…

**Comment by coreyfloyd** [comment]
@bgerstle can i get a re-review here?

**Comment by bgerstle** [comment]
@coreyfloyd i can re-run to see if the bug i noticed is fixed, but i'm not sure why removing the index dictionary fixes the issue. would be helpful to see a test that demonstrates the problem.  might not get to it until later tonight or tomorrow morning

**Comment by bgerstle** [comment]
bug doesn't happen anymore, but i'm still wondering what the issue was... maybe we weren't updating the index in certain cases. anyway, fine to see the pre-optimization go, we can address if performance becomes a problem later. :+1:

**Pull request merged**